hawkular-inventory-paths/src/main/java/org/hawkular/inventory/paths/ElementTypeVisitor.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.inventory.paths;
+
+import org.hawkular.inventory.paths.Path.SegmentType;
+
+/**
+ * A visitor interface to accept different kinds of entities available in Hawkular.
+ *
+ * @author Lukas Krejci
+ * @since 0.1.0
+ */
+public interface ElementTypeVisitor<R, P> {
```

Comment:
> I think you moved this over because it is used somewhere in the *Path impls,

Yes.

> but I'm not sure I like this. This would have a huge impact on the rest of the inventory code base where this guy is used a lot.

Yes, it has an impact, but I would not say it is huge. See e057fde - would that be acceptable?

> We can either rename this to SegmentTypeVisitor and essentially copy the "true" ElementTypeVisitor, or we can add some conversion function somewhere (probably in inventory) that would convert an element class to a segment type and use this throughout the codebase.

I added some overloads to path related classes that accept `Class<?>` so that the impact in the rest of Inventory is smaller, e.g. here https://github.com/ppalaga/hawkular-inventory/commit/e057fde5ec7749de54df9f72228f97a551b2bfc4#diff-1afa9c36e40cd992929467498f761a59R128

If you still think `SegmentTypeVisitor` can be cut out from *paths I'd rather hand this over to you 'cause I do not feel to know enough `SegmentTypeVisitor` and how it is used.

> A third option is to move a majority of org.hawkular.inventory.api over to this new place, but I like that the least because the API evolves and having it and inv split in 2 repos is a little bit impractical.

-1